File: src/main/org/nlogo/prim/gui/_updatemonitor.java
package org.nlogo.prim.gui;

import org.nlogo.api.LogoException;
import org.nlogo.nvm.Context;
import org.nlogo.nvm.Syntax;

public final strictfp class _updatemonitor
    extends org.nlogo.nvm.Command {
  @Override
  public Syntax syntax() {
    return Syntax.commandSyntax
        (new int[]{Syntax.TYPE_WILDCARD},
            "O---", true);
  }

  @Override
  public void perform(final Context context)
      throws LogoException {
    ((org.nlogo.window.MonitorWidget) context.job.owner)
        .value(args[0].report(context));
    context.ip = next;
  }
}

File: src/main/org/nlogo/prim/gui/_updatemonitor.scala
package org.nlogo.prim.gui

import org.nlogo.api.LogoException
import org.nlogo.nvm.{ Command, Context, Syntax }
import org.nlogo.window.MonitorWidget

class _updatemonitor extends Command {
  override def syntax =
    Syntax.commandSyntax(
      Array(Syntax.TYPE_WILDCARD), "O---", true);
  override def perform(context: Context) {
    context.job.owner.asInstanceOf[MonitorWidget]
      .value(args(0).report(context))
    context.ip = next
  }
}
